assets/src/scss/layout/navbar/_burger-menu.scss

```diff
@@ -0,0 +1,268 @@
+#burger-menu {
+  _-- {
+    background: $white;
+    color: $grey-80;
+    font-family: $roboto;
+  }
+
+  position: fixed;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  top: 0;
+  left: 0;
+  inset-inline-start: -100%;
+  padding: 18px $sp-3;
+  z-index: 99999;
+  overflow: hidden;
+  height: 100%;
+
+  &.open {
+    inset-inline-start: 0;
+    transition: inset-inline-start 0.2s ease-out;
+
+    @supports not (inset-inline-start: 0) {
+      transform: translateX(0);
+      transition: transform 0.2s ease-out;
+
+      html[dir="rtl"] & {
+        transform: translateX(0);
+      }
+    }
+
+    ~ .burger-menu-overlay {
+      display: block;
+
+      @include large-and-up {
+        display: none;
+      }
+    }
+  }
+
+  ul {
+    padding: 0;
+    list-style: none;
+  }
+
+  a.nav-link {
+    color: $grey-80;
+    width: auto;
+  }
+
+  .admin-bar & {
+    @include medium-and-less {
+      top: 46px;
+      height: calc(100% - 46px);
+    }
+
+    @media (min-width: 783px) {
+      top: 32px;
+      height: calc(100% - 32px);
+    }
+  }
+
+  @include small-and-less {
+    width: 100vw;
+    inset-inline-start: -100vw;
+
+    @supports not (inset-inline-start: -100vw) {
+      transform: translateX(-100vw);
+      transition: transform 0.2s ease-in;
+
+      html[dir="rtl"] & {
+        transform: translateX(100vw);
+      }
+    }
+  }
+
+  @include small-to-large {
+    width: 400px;
+    inset-inline-start: -400px;
+    transition: inset-inline-start 0.2s ease-in;
+
+    @supports not (inset-inline-start: -400px) {
+      transform: translateX(-400px);
+      transition: transform 0.2s ease-in;
+
+      html[dir="rtl"] & {
+        transform: translateX(400px);
+      }
+    }
+  }
+
+  @include large-and-up {
+    display: none;
+  }
+}
+
+#burger-menu .burger-menu-header {
```

Comment:
If we provide our markup with very specific class names that completely include the ID of the parent element `burger-menu`, it's not needed to add this ID in CSS before every such class.